relay-sampling/src/lib.rs

```diff
@@ -197,11 +176,8 @@ pub struct CustomCondition {
 }
 
 impl CustomCondition {
-    fn matches_event(&self, event: &Event, ip_addr: Option<IpAddr>) -> bool {
-        Event::get_custom_operator(&self.name)(self, event, ip_addr)
-    }
-    fn matches_trace(&self, trace: &DynamicSamplingContext, ip_addr: Option<IpAddr>) -> bool {
-        DynamicSamplingContext::get_custom_operator(&self.name)(self, trace, ip_addr)
+    fn matches<T: FieldValueProvider>(&self, value_provider: &T, ip_addr: Option<IpAddr>) -> bool {
```

Comment:
_nit_: While we're at it, could we rewrite this into the more verbose but also more consistent style? This is mostly my personal preference, but it also improves readability in most cases.

```suggestion
    fn matches<T>(&self, value_provider: &T, ip_addr: Option<IpAddr>) -> bool
    where
        T: FieldValueProvider
    {
```